src/ConcordiumNetSdk/Types/Hash.cs

```diff
@@ -1,58 +1,70 @@
 namespace ConcordiumNetSdk.Types;
 
 /// <summary>
-/// Represents a base16 encoded hash.
+/// Represents a hash.
 /// </summary>
 public abstract class Hash
 {
     private const int StringLength = 64;
     public const int BytesLength = 32;
 
+    /// <summary>
+    /// A lowercase length-64 hex encoded string representing the hash.
+    /// </summary>
     private readonly string _formatted;
```

Comment:
Please don't do this. A hash is 32 bytes. This hex is just one rendering of it. We don't need to store both representations internally.